onnxruntime/core/framework/tensorprotoutils.cc

```diff
@@ -165,23 +165,57 @@ static Status GetExternalDataInfo(const ONNX_NAMESPACE::TensorProto& tensor_prot
 // This function does not unpack string_data of an initializer tensor
 Status ReadExternalDataForTensor(const ONNX_NAMESPACE::TensorProto& tensor_proto,
                                  const ORTCHAR_T* tensor_proto_dir,
-                                 std::vector<uint8_t>& unpacked_tensor) {
+                                 std::unique_ptr<uint8_t[]>& unpacked_tensor, size_t& tensor_byte_size) {
   std::basic_string<ORTCHAR_T> external_file_path;
   onnxruntime::FileOffsetType file_offset;
-  SafeInt<size_t> tensor_byte_size;
+  SafeInt<size_t> safe_tensor_byte_size;
   ORT_RETURN_IF_ERROR(GetExternalDataInfo(
       tensor_proto,
       tensor_proto_dir,
       external_file_path,
       file_offset,
-      tensor_byte_size));
+      safe_tensor_byte_size));
+  tensor_byte_size = safe_tensor_byte_size;
 
-  unpacked_tensor.resize(tensor_byte_size);
+  unpacked_tensor = std::unique_ptr<uint8_t[]>(new uint8_t[tensor_byte_size]); // std::make_unique_for_overwrite<uint8_t[]>(tensor_byte_size);
   ORT_RETURN_IF_ERROR(onnxruntime::Env::Default().ReadFileIntoBuffer(
       external_file_path.c_str(),
       file_offset,
       tensor_byte_size,
-      gsl::make_span(reinterpret_cast<char*>(unpacked_tensor.data()), tensor_byte_size)));
+      gsl::make_span(reinterpret_cast<char*>(unpacked_tensor.get()), tensor_byte_size)));
+
+  return Status::OK();
+}
+
+// Read external data for tensor in unint8_t* form and return Status::OK() if the data is read successfully.
+// Uses the tensor_proto_dir to construct the full path for external data. If tensor_proto_dir == nullptr
+// then uses the current directory instead.
+// This function does not unpack string_data of an initializer tensor
+Status ReadExternalDataForTensor(const ONNX_NAMESPACE::TensorProto& tensor_proto,
```

Comment:
Does this function need the overflow vector? It feels like the only place this function is used in this file is with a known byte size, and I'm not sure we should silently succeed if the expected byte size doesn't match the actual byte size. Maybe there's something I'm missing here (e.g. alignment, padding)?

Reply:
I don't think so, everything seems to make sense. I started a CI run.